src/errors/ledger.errors.ts

```diff
@@ -0,0 +1,78 @@
+import { ICP } from "../icp";
+import { BlockHeight } from "../types/common";
+
+export type TransferError =
```

Comment:
Wouldn't it make more sense to create this as a class Error and the others inherit from this so that we could check it at run time if needed?

Reply:
Definitely. I transformed it in `abstract class`, thanks.